hyperspy/tests/signal/test_spike_removal_tool.py

```diff
@@ -58,3 +58,19 @@ def test_spikes_removal_tool():
     sr.apply()
     np.testing.assert_almost_equal(s.data[1, 2, 14], 1, decimal=5)
     assert s.axes_manager.indices == (0, 0)
+
+
+def test_spikes_removal_tool_non_interactive():
+    s = Signal1D(np.ones((2, 3, 30)))
+    np.random.seed(1)
+    s.add_gaussian_noise(1e-5)
+    # Add three spikes
+    s.data[1, 0, 1] += 2
+    s.data[0, 2, 29] += 1
+    s.data[1, 2, 14] += 1
+
+    sr = s.spikes_removal_tool(threshold=0.5, interactive=False, add_noise=False)
+    np.testing.assert_almost_equal(s.data[1, 0, 1], 1, decimal=5)
+    np.testing.assert_almost_equal(s.data[0, 2, 29], 1, decimal=5)
+    np.testing.assert_almost_equal(s.data[1, 2, 14], 1, decimal=5)
+    assert isinstance(sr, SpikesRemoval)
```

Comment:
Add tests for masking in the non_interactive mode?

`add_noise` could be tested both `True/False` using `@pytest.mark.parametrize(("add_noise"), (True, False))`

Reply:
Implemented using `@parametrize`.